pxr/usdImaging/hdEmscripten/js/ThreeJsRenderDelegate.js

```diff
@@ -0,0 +1,430 @@
+class TextureRegistry {
+  constructor(basename) {
+    this.basename = basename;
+    this.textures = [];
+    this.loader = new THREE.TextureLoader();
+  }
+  getTexture(filename) {
+    if (this.textures[filename]) {
+      return this.textures[filename];
+    }
+
+    let textureResolve, textureReject;
+    this.textures[filename] = new Promise((resolve, reject) => {
+      textureResolve = resolve;
+      textureReject = reject;
+    });
+
+    let resourcePath = filename;
+    if (filename[0] !== '/') {
+      resourcePath = this.basename + '[' + filename +']';
+    }
+
+    let filetype = undefined;
+    if (filename.indexOf('.png') >= filename.length - 5) {
+      filetype = 'image/png';
+    } else if (filename.indexOf('.jpg') >= filename.length - 5) {
+      filetype = 'image/jpeg';
+    } else if (filename.indexOf('.jpeg') >= filename.length - 5) {
+      filetype = 'image/jpeg';
+    } else {
+      throw new Error('Unknown filetype');
+    }

```

Comment:
You likely want to do something like this instead:
```suggestion
let filetype = undefined;

 if (REGEX_JPEG.test(filename)) {
  filetype = 'image/jpeg';
} else if(REGEX_PNG.test(filename)) {
  filetype = 'image/png';
} else {
  throw new Error('Unknown filetype');
}
```

Where at the top of the file you do:
```
const REGEX_JPEG = /\.(jpg|jpeg)$/i;
const REGEX_PNG = /\.png$/i;
```